pollux/lib/vc-jwt/src/main/scala/io/iohk/atala/pollux/vc/jwt/Proof.scala

```diff
@@ -0,0 +1,110 @@
+package io.iohk.atala.pollux.vc.jwt
+
+import com.nimbusds.jose.crypto.bc.BouncyCastleProviderSingleton
+import io.circe.*
+import io.circe.syntax.*
+
+import java.time.{Instant, ZoneOffset}
+import zio.*
+import io.iohk.atala.shared.utils.Json as JsonUtils
+import io.iohk.atala.shared.utils.Base64Utils
+import scodec.bits.ByteVector
+import java.security.*
+
+sealed trait Proof {
+  val id: Option[String] = None
+  val `type`: String
+  val proofPurpose: String
+  val verificationMethod: String
+  val created: Option[Instant] = None
+  val domain: Option[String] = None
+  val challenge: Option[String] = None
+  val proofValue: String
+  val previousProof: Option[String] = None
+  val nonce: Option[String] = None
+}
+
+trait EddsaJcs2022ProofGenerator {
+  private val provider = BouncyCastleProviderSingleton.getInstance
+  def generateProof(payload: Json, sk: PrivateKey, pk: PublicKey): Task[EddsaJcs2022Proof] = {
```

Comment:
Do we have unit tests validating the interoperability of our implementation? I guess this is something we'd like to cover?

Reply:
I was more thinking about interoperability testing. That is, making sure what are able to verify a proof generated by another system and making sure the ones we create are standard compliant. It could, for instance, be a unit-test verifying a hardcoded proof generated by another system.

I think it is important given we implemented the proof aggregation/signing/C14N code ourself.